cmd/buildah/export.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containers/storage/pkg/archive"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/buildah"
+	"github.com/urfave/cli"
+)
+
+var (
+	exportFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "output, o",
+			Usage: "write to a file, instead of STDOUT",
+		},
+	}
+	exportCommand = cli.Command{
+		Name:  "export",
+		Usage: "Export the contents of a container's filesystem as a tar archive",
+		Description: `Export the contents of a container's filesystem using the full or shortened container ID or container name. The output is exported to STDOUT and can be redirected to a tar file.
+
+   Stream to a file instead of STDOUT by using -o.
+       `,
+		Flags:     exportFlags,
+		Action:    exportCmd,
+		ArgsUsage: "CONTAINER",
+	}
+)
+
+func exportCmd(c *cli.Context) error {
+	var builder *buildah.Builder
+
+	args := c.Args()
+	if len(args) == 0 {
+		return errors.Errorf("container name must be specified")
+	}
+	if len(args) > 1 {
+		return errors.Errorf("too many arguments specified")
+	}
+
+	outfile := "/dev/stdout"
+	if c.IsSet("output") {
+		outfile = c.String("output")
+	} else {
+		stat, _ := os.Stdout.Stat()
+		if (stat.Mode() & os.ModeCharDevice) != 0 {
+			return errors.Errorf("Refusing to save to a terminal. Use the -o flag or redirect.")
+		}
```

Comment:
We use github.com/Sirupsen/logrus.IsTerminal() elsewhere, so it might as well get used here, too.